crates/types/src/nodes_config.rs

```diff
@@ -57,15 +60,77 @@ pub enum Role {
 #[display(fmt = "v{}", _0)]
 pub struct ConfigVersion(u32);
 
+impl ConfigVersion {
+    pub const INVALID: ConfigVersion = ConfigVersion(0);
+    pub const MIN: ConfigVersion = ConfigVersion(1);
+}
+
+impl Default for ConfigVersion {
+    fn default() -> Self {
+        Self::MIN
+    }
+}
+
 #[derive(Debug, Clone, Eq, PartialEq, Default)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-struct NodesConfiguration {
+pub struct NodesConfiguration {
     version: ConfigVersion,
     nodes: HashMap<PlainNodeId, MaybeNode>,
     controllers: Vec<ControllerConfig>,
     name_lookup: HashMap<String, PlainNodeId>,
 }
 
+impl NodesConfiguration {
+    /// The currently loaded nodes configuration. This returns None if no configuration
+    /// is loaded yet, this will be only the case before attaching to the controller
+    /// node on startup.
+    ///
+    /// After attaching, it's safe to unwrap this, or use current_unchecked() variant.
+    pub fn current() -> Option<Arc<NodesConfiguration>> {
+        CURRENT_CONFIG.load_full()
+    }
+
+    pub fn current_unchecked() -> Arc<NodesConfiguration> {
+        CURRENT_CONFIG.load_full().unwrap()
+    }
+
+    /// The currently loaded configuration version or ConfigVersion::INVALID if
+    /// no configuration is loaded yet.
+    pub fn current_version() -> ConfigVersion {
+        CURRENT_CONFIG
+            .load()
+            .as_ref()
+            .map(|config| config.version)
+            .unwrap_or(ConfigVersion::INVALID)
+    }
+
+    /// Silently ignore the config if it's older than existing.
+    /// Note that this is not meant to be called concurrently from multiple tasks, the system have
+    /// a single task responsible for setting updates.
+    pub fn set_as_current_if_newer(self) {
+        let current = CURRENT_CONFIG.load_full();
+        match current {
+            None => {
+                CURRENT_CONFIG.compare_and_swap(&current, Some(Arc::new(self)));
+            }
+            Some(current) if self.version > current.version => {
+                CURRENT_CONFIG.compare_and_swap(&current, Some(Arc::new(self)));
+            }
+            Some(_) => { /* Do nothing, current is already newer */ }
+        }
+    }
```

Comment:
If this method is not meant to be used concurrently, can we replace `compare_and_swap` with `store`? `compare_and_swap` kind of gives the impression to me that something concurrently could happen.